core/controllers/editor.py

```diff
@@ -775,3 +777,51 @@ def get(self, exploration_id):
         self.render_json({
             'unresolved_answers': unresolved_answers_with_frequency
         })
+
+
+class LearnerAnswerInfoHandler(EditorHandler):
+    """Handles the learner answer info for an exploration state."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_edit_exploration
+    def get(self, exploration_id):
+        """Handles the GET requests for for learner answer info for an
+        exploration state.
+        """
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        state_name = self.request.get('state_name')
+        if not state_name:
+            raise self.PageNotFoundException
+
+        entity_type = feconf.ENTITY_TYPE_EXPLORATION
+        state_reference = (
+            stats_models.LearnerAnswerDetailsModel
```

Comment:
I think you should write this in a service layer. Controllers should call services not models directly.

Reply:
I checked that at many places we have used models functions in the controllers file. By the way its just generating state reference.

Reply:
I'm not sure which file you are talking about, but is there any issue with moving this to the service layer?

Reply:
I concur with @DubeySandeep. Please move this to the service layer.

Reply:
Done

Reply:
@anubhavsinha98, I don't think this is "Done!", can you please check?